Render teammate component on proxy

CodeWritingCow opened the pull request:
Please review my code. Thanks!

CodeWritingCow committed:
Create proxy server

Server loads bundle.js containing React components. However, there is a bug.

Although the proxy’s index.html renders <Category> components
from bundle.js, the components are throwing errors.

The components are not getting data from the API server. It's because
they are making GET requests to localhost:3017 (the proxy) rather than
localhost:3016 (the API server).

Since <Category>s are not getting any data from the API, they are not
rendering any <Event> component.

CodeWritingCow committed:
Comment out ReactDOM.render on proxy page

Since bundle.js contains React components already transpiled into vanilla JavaScript, it is unnecessary to run ReactDOM.render() in index.html.

CodeWritingCow committed:
Render teammate's component with proxy

Proxy server can now render Grahn's component.

WormJim commented on server/app.js:
```diff
@@ -0,0 +1,19 @@
+app.get('/', (req, res, err) => {
+    res.send(data);
+});
+
```
This Code above doesn't do anything special.  

If your intention is to send the public directory with all it's content, the code `app.use(express.static(__dirname + '/../public'));` is doing just that.  However, your path should be that you specify here isn't configured correctly.

When using the above syntax, and appending the path `'/../public'` to the `__dirname'  you are effectively extending the path to the `sever.js` file to `server.js/../public`.  which is not what you want.  You need to use the path module provided by Node.

WormJim replied:
Ahh, you changed it.  Great!!!

WormJim commented on server/app.js:
```diff
@@ -0,0 +1,19 @@
+    console.log(`Listening on port ${port}`);    
+});
+
+module.exports = app;
```
A single page application will need to send back the proxy index.html file every time the path in the URL is changed.  
The way this proxy is set up will never handle any of the React Components JavaScript. This is a "dumb" proxy which should only renders the main `index.html` file.  Create a path for any route that you expect to be handled, which in turn returns that file.

If you'd like to use a Route in express to handle any direction to `/:id` there are two ways to achieve this.

1. create an `app.get` route to handle this path.  The route must return the index.html file
2. create a "Catch All" route which effectively will catch any path not specifically declared. The route must return the index.html file.  You can use the asterisk as a catch all. Example: `app.get('*', callback)`

WormJim commented on public/index.html:
```diff
@@ -14,8 +14,8 @@ <h1>Proxy Server</h1>
         ReactDOM.render(React.createElement(App), document.getElementById('app'));
-    </script>
+    </script> -->
 </body>
```
Very Good catch. Get rid of the RectDom Render function.  Not needed.

WormJim commented on server/app.js:
```diff
@@ -1,13 +1,14 @@
 app.use(express.static(__dirname + '/../public'));
-// app.use(express.static(__dirname + '/../client/dist'));
+// app.use(express.static(path.join(__dirname, 'public')));
 
```
This is great.  This is exactly what I was referring to in an earlier comment.  You can disregard that comment.

WormJim commented on public/index.html:
```diff
@@ -3,19 +3,17 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
-    <title>Page Title</title>
+    <title>Proxy Page</title>
     <meta name="viewport" content="width=device-width, initial-scale=1">
```
The title should emulate Trip Advisor.  That is what we are trying to create.

WormJim commented on public/index.html:
```diff
@@ -3,19 +3,17 @@
-    <div id="app"></div>
+    <div id="Details"></div>
+    <div id="events"></div>
     <script src="/lib/react.development.js"></script>
```
Let's be consistent here.  I'd suggest making the id's Proper Case.  That way they are easily recognized. 

Also, I recall you using the word 'Categories' before.  I think that best explains what your component is and does.  It renders additional Categories of Attractions the purchaser may be interested in.

CodeWritingCow replied:
Thanks. Done!

WormJim commented on public/index.html:
```diff
@@ -3,19 +3,17 @@
-    <!-- <script src="http://localhost:3016/client/dist/bundle.js"></script> -->
+    <script src="http://localhost:3014/bundle.js"></script>
     <script src="http://localhost:3016/bundle.js"></script>
-    <!-- <script>
```
Awesome that you removed the actual path to the bundle.js file.  Can you tell me why you don't need to use that?

Hint: it has something to do with `express.static()`

CodeWritingCow committed:
Integrate Sidebar component with proxy

Add Grahn's Sidebar to proxy. Also, change div ID 'events' to 'Categories.'

TODO: Stylize Sidebar with CSS.

CodeWritingCow committed:
Change localhost URL to AWS IP

Pull bundle.js and CSS stylesheet from AWS instances instead of localhosts.
